### PR TITLE
[ONNX] Move symbolic export to separate file

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #95935
* #95929
* #95676
* #95664
* #95651
* #95686
* __->__ #95650

Move things around in the effort of preparing to refactor
the code structure.

### CODE REVIEW

File: torch/onnx/_internal/fx/symbolic_exporter.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+            # , which is not valid Python code.
+            node.args = (None,)
+
+            replaced_attrs.append(replaced_attr)
+
+    return tuple(replaced_attrs)
+
+
+@_beartype.beartype
+def _trace_into_fx_graph_via_fx_symbolic_trace(
```
Comment (wschin) [inline torch/onnx/_internal/fx/symbolic_exporter.py:131]: Make it a FX frontend?

File: torch/onnx/_internal/fx/symbolic_exporter.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+    @_beartype.beartype
+    def to_bool(self, obj: "torch.fx.Proxy") -> bool:
+        # FIXME: This is a hack to tracing through if-else Python blocks.
+        # It may generate incorrect ONNX graphs if the if-else block
+        return False
+
+
+@_beartype.beartype
+def _move_placeholder_to_front(graph_module: torch.fx.GraphModule) -> None:
```
Comment (wschin) [inline torch/onnx/_internal/fx/symbolic_exporter.py:60]: There are some graph transforms in this file. They are good candidates for new `Transform` classes.

Comment (BowenBao) [reply torch/onnx/_internal/fx/symbolic_exporter.py:60]: Done in the pass PR.

**Review by titaiwangms — APPROVED** [review]

File: torch/onnx/_internal/fx/symbolic_exporter.py (hunk 1 above)
Comment (BowenBao) [reply torch/onnx/_internal/fx/symbolic_exporter.py:131]: Will attempt in a follow up.

**Pull request closed without merging**